frontend/src/pages/IstioConfigNew/GatewayForm/ListenerBuilder.tsx

```diff
@@ -78,83 +99,140 @@ export const ListenerBuilder: React.FC<ListenerBuilderProps> = (props: ListenerB
-          {allowedRoutes.map((option, index) => (
-            <FormSelectOption isDisabled={false} key={`p_${index}`} value={option} label={option} />
-          ))}
-        </FormSelect>
-      </Td>
-
-      <Td>
-        <TextInput
-          id={`addSelectorLabels_${props.index}`}
-          name="addSelectorLabels"
-          onChange={onAddSelectorLabels}
-          validated={isValid(isValidSelector(props.listener.sSelectorLabels))}
-        />
-      </Td>
-
-      <Td>
-        <Button
-          id={`deleteBtn_${props.index}`}
-          variant={ButtonVariant.link}
-          icon={<KialiIcon.Trash />}
-          style={{ padding: 0 }}
-          onClick={() => props.onRemoveListener(props.index)}
-        />
-      </Td>
-    </Tr>
+    <>
+      <Tr>
+        <Td>
+          <TextInput
+            value={props.listener.name}
+            type="text"
+            id={`addName_${props.index}`}
+            aria-describedby="add name"
+            onChange={onAddName}
+            validated={isValid(isValidName(props.listener.name))}
+          />
+        </Td>
+
+        <Td>
+          <TextInput
+            value={props.listener.hostname}
+            type="text"
+            id={`addHostname_${props.index}`}
+            aria-describedby="add hostname"
+            name="addHostname"
+            onChange={onAddHostname}
+            validated={isValid(isValidHostname(props.listener.hostname))}
+          />
+        </Td>
+
+        <Td>
+          <TextInput
+            value={props.listener.port}
+            type="text"
+            id={`addPort_${props.index}`}
+            placeholder="80"
+            aria-describedby="add port"
+            name="addPortNumber"
+            onChange={onAddPort}
+            validated={isValid(isValidPort(props.listener.port))}
+          />
+        </Td>
+
+        <Td>
+          <FormSelect
+            value={props.listener.protocol}
+            id={`addPortProtocol_${props.index}`}
+            name="addPortProtocol"
+            onChange={onAddProtocol}
+          >
+            {protocols.map((option, index) => (
+              <FormSelectOption isDisabled={false} key={`p_${index}`} value={option} label={option} />
+            ))}
+          </FormSelect>
+        </Td>
+
+        <Td>
+          <FormSelect value={props.listener.from} id={`addFrom_${props.index}`} name="addFrom" onChange={onAddFrom}>
+            {allowedRoutes.map((option, index) => (
+              <FormSelectOption isDisabled={false} key={`p_${index}`} value={option} label={option} />
+            ))}
+          </FormSelect>
+        </Td>
+
+        <Td>
+          <TextInput
+            id={`addSelectorLabels_${props.index}`}
+            name="addSelectorLabels"
+            isDisabled={props.listener.from !== SELECTOR}
+            onChange={onAddSelectorLabels}
+            validated={isValid(isValidSelector(props.listener.sSelectorLabels))}
```

Comment:
```suggestion
            validated={isValid(
              props.listener.from === SELECTOR ? isValidSelector(props.listener.sSelectorLabels) : undefined
            )}
```
Apply validation only if the text input is enabled